kernel/include/kernel_arch.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2016 Wind River Systems, Inc.
+ * Copyright (c) 2019 Stephanos Ioannidis <root@stephanos.io>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_KERNEL_INCLUDE_KERNEL_ARCH_H_
```

Comment:
why did you decide to call this kernel_arch.h?

What is the intended purpose of this header? I'd rather you called this something else, we already have headers for the kernel code to arch interface. It also appears you have some duplicate definitions in here.

You *can* however, move include/sys/arch_interface.h to kernel/arch_interface.h if you like.

Reply:
kernel_arch.h is really just something I added to achieve what this PR intends to do (remove all the junk from kernel_structs.h so that it can be included in arch headers), with the least amount of refactoring involved.

I know this may not be the best solution in terms of the long term header clean-up strategy, but at least for now, this seems to be a prominent solution for achieving the main goal of this PR.

I will review #19666 and see if I can come up with something better tomorrow.